### PR TITLE
Fix the notification favicon

### PR DESCRIPTION
**Pull request by ChristophWurst**
The code to our special notification icon with the red dot was
somehow broken. This triggers the favicon change even again and
also fixes the outdated jQuery selector for the favicon DOM element.

Fixes https://github.com/nextcloud/mail/issues/406

cc @fsedarkalex

### CODE REVIEW

**Comment by skjnldsv** [comment]
Me neither.
@ChristophWurst, we're supposed to get the same number on the icon as the number in the page title?

**Comment by ChristophWurst** [comment]
Retested and works for me: 
bildschirmfoto von 2017-09-06 09-53-39

Note that you have to change tabs / application to get an updated notification favicon. We only show it if the window is not focused.

**Comment by ChristophWurst** [comment]
Okay, I found a little flaw. The favicon was not updated if one didn't allow desktop notifications. Now it updates it regardless of the notification permissions :)

**Comment by jancborchardt** [comment]
Hm, still doesn’t work for me in neither Firefox nor Chromium.

How about you @fsedarkalex @lcalaresu @Gomez?

**Comment by ChristophWurst** [comment]
> Hm, still doesn’t work for me in neither Firefox nor Chromium.
> 
> How about you @fsedarkalex @lcalaresu @Gomez?

Please provide the *exact* steps you took to test this. Did you run `make optimize-js`? Did you refresh your browser window? ;)

**Comment by ChristophWurst** [comment]
> Please provide the exact steps you took to test this. Did you run make optimize-js? Did you refresh your browser window? ;)

And most importantly: did you receive any new emails while the browser tab was not in focus? Did you see a desktop notification?

**Comment by skjnldsv** [comment]
@ChristophWurst the mails are not refreshed if not focused, that's why I can't test it!
No requests in the network console either. :/

**Comment by jancborchardt** [comment]
@ChristophWurst I followed all these steps. I’m running on this branch, ran the make optimize-js, sent a mail to myself, changed to another tab, and I’m also getting the notifications. But no red dot. I tested in Firefox and Chromium and also in private tabs.

**Comment by skjnldsv** [comment]
ToFix: go back to the old favicon (no red dot) on window focus?

**Comment by ChristophWurst** [comment]
> ToFix: go back to the old favicon (no red dot) on window focus?

Should be good now, please re-test one more time ;)

**Comment by jancborchardt** [comment]
A possible additonal enhancement seems to be that when you read the mail on another device, that the next background refresh will remove the red dot from the favicon. Would that work @ChristophWurst?

**Comment by ChristophWurst** [comment]
A bit tricky but definitely possible.

**Comment by ChristophWurst** [comment]
As always, pleas open a ticket so that we don't forget about it.

File: js/views/appview.js [hunk 1]
```diff
@@ -120,8 +120,9 @@ define(function(require) {
 			Radio.ui.trigger('document:click', event);
 		},
 		onDocumentShow: function(e) {
+			console.error('SHOW');
```
Comment (nickvergessen) [inline js/views/appview.js]: 🙈

Comment (ChristophWurst) [reply js/views/appview.js]: haha, whoops

Comment (skjnldsv) [reply js/views/appview.js]: I though it was on purpose :p

**Comment by jancborchardt** [comment]
Follow-up issue at https://github.com/nextcloud/mail/issues/507 :)

**Comment by fsedarkalex** [comment]
When talking about favicon are we actually talking about the icon on the browser tab?
Am I right it should also be displayed with red dot when I am currently e.g. in contacts app?
That is what I was especially talking about.
Further I think the dot should also be visible next to mail app's icon in the header or menu list or something like that...

My results:
The red dot appears but only when mail app is active.

My additional Question:
When I am on mail apps tab I see a number of unread mails in the title. When I am somewhere else I don't. Why? Bug? :)

**Comment by ChristophWurst** [comment]
> When talking about favicon are we actually talking about the icon on the browser tab?
> Am I right it should also be displayed with red dot when I am currently e.g. in contacts app?
> That is what I was especially talking about.
> Further I think the dot should also be visible next to mail app's icon in the header or menu list or something like that...

Yes, we only update the tab's favicon and not the app icon in the app menu. Updating the app icon also from other apps would mean we'd have to poll for new messages in any opened tab. That would require a lot of processing power and is not worth the additional complexity it would introduce into this app. But still, feel free to open an enhancement ticket ;)

> The red dot appears but only when mail app is active.

It appears if the app is *not* active.

> When I am on mail apps tab I see a number of unread mails in the title. When I am somewhere else (meaning when the red dot appears) I don't. Why? Bug? :)

Probably. I think we don't synchronize the number of unread messages when we sync new/changed/vanished messages in background. Please file an issue and I'll take a look at this.

**Comment by fsedarkalex** [comment]
> > The red dot appears but only when mail app is active.
> It appears if the app is not active.
we should find some better way to say what we are talking about :)

"app active" = clicked mail app
"app inactive" = clicked e.g. contacts app

what you mean with app inactive is using another browser tab, right?

I will open two more "issues" :)

**Comment by ChristophWurst** [comment]
> what you mean with app inactive is using another browser tab, right?

Exactly :)

**Pull request merged**